### PR TITLE
feat: scrape multiple years and sessions

### LINKED ISSUE #21: Support multiple years and semesters
- [ ] Scrape data from multiple endpoints (currently hardcoded to `/sws2021/`) into multiple JSON files. Past years are available.
- [x] Select source in UI - dropdown next to search with default to current year?

### PR DESCRIPTION
**Pull request by pl4nty**
* Scraper accepts commandline input
* Scrape Action uses matrix of years and sessions
* Fixes #21

### CODE REVIEW

**Comment by pl4nty** [comment]
I might not be much of a dev, but ops on the other hand :P

**Comment by pl4nty** [comment]
Results: https://github.com/anucssa/anutimetable/actions/runs/1703121838

**Comment by pl4nty** [comment]
And it's fixed, apologies for the pre-emptive review request. I might see if 2021 works too, just out of interest

File: .github/workflows/scrape.yml [hunk 1]
```diff
@@ -9,6 +9,13 @@ jobs:
   scrape:
     runs-on: ubuntu-latest
     name: Scrape data into JSON file with Python script
+    env:
+      TIMETABLE_PREFIX: http://timetabling.anu.edu.au/sws
+    strategy:
+      fail-fast: false
+      matrix:
+        year: [2022]
+        session: ['1 S1', '2 S2', '3 X1', '4 X2', '5 X3', '6 X4']
```
Comment (OliverBalfour) [inline .github/workflows/scrape.yml:18]: What happens if one of the combinations in future doesn't exist? Is that what `fail-fast: false` is there for, to make it silently fail and continue with other combos?

File: .github/workflows/scrape.yml [hunk 2]
```diff
@@ -40,7 +54,7 @@ jobs:
           GH_PAT: ${{ secrets.GH_PAT }}
         run: |
           cd scraper
-          mv -f ./timetable.json ../public/
+          mv -f ./timetable.json ../public/timetable_${YEAR}_${SESSION#* }.json
```
Comment (OliverBalfour) [inline .github/workflows/scrape.yml:60]: Had to do a google search to grok this, a comment might help. To my understanding this is getting the part of `SESSION` after the first space, eg `S1` from `1 S1`, correct?

**Comment by OliverBalfour** [comment]
> I might not be much of a dev

timetable.cssa.club says otherwise 😛

File: .github/workflows/scrape.yml (hunk 2 above)
Comment (pl4nty) [reply .github/workflows/scrape.yml:60]: Oh sorry, yes I was supposed to put a comment here

File: .github/workflows/scrape.yml (hunk 1 above)
Comment (pl4nty) [reply .github/workflows/scrape.yml:18]: > Spot on

Huh not sure why my comment (above) didn't show up

**Comment by pl4nty** [comment]
If you're happy, I can merge once this passes https://github.com/anucssa/anutimetable/actions/runs/1703168875

**Comment by pl4nty** [comment]
Ah, I wondered if we could hit that race condition

**Pull request merged**